### PR TITLE
Add list of tests that work with Py_LIMITED_API

### PR DESCRIPTION
**Pull request by da-woods**
Apply it to a few tests, and do some small limited api fixes.

This is mainly intended to make it easier to test more stuff - the fixes here are pretty minor.

### CODE REVIEW

File: Cython/Utility/TypeConversion.c [hunk 1]
```diff
@@ -1,8 +1,12 @@
 /////////////// TypeConversions.proto ///////////////
-//@requires: StringTools.c::IncludeStringH
 
 /* Type Conversion Predeclarations */
 
+// Note that this is *always* included in the module preamble
+// (but the impl part is only generated on demand).
+// Therefore IncludeStringH is not be reliable
+#include <string.h>
```
Comment (scoder) [inline Cython/Utility/TypeConversion.c:8]: I've started writing a cleanup to at least avoid needing this in the `.proto` part. It should be enough to include it in the impl section. We can then exclude these changes from this PR.

File: Cython/Utility/ModuleSetupCode.c [hunk 2]
```diff
@@ -1,6 +1,10 @@
 /////////////// InitLimitedAPI ///////////////
 
-#if defined(CYTHON_LIMITED_API) && 0  /* disabled: enabling Py_LIMITED_API needs more work */
+#if defined(CYTHON_LIMITED_API) && defined(CYTHON_TEST_IN_LIMITED_API)
+  /* disabled by default: enabling Py_LIMITED_API needs more work.
+     Users should not define CYTHON_TEST_IN_LIMITED_API since it will
+     be removed when support is better.
+     If you want Py_LIMITED_API then set it yourself for now */
```
Comment (scoder) [inline Cython/Utility/ModuleSetupCode.c:7]: Hmm, then why don't *we*  set `Py_LIMITED_API` in the tests and instead need a new macro guard? The `CYTHON_*` macro names are specifically meant for users to define. I would leave it to `runtests.py` to set `Py_LIMITED_API` to something reasonable.

**Review by scoder — COMMENTED** [review]
I'm not so happy that we add a new tag that means "this test works in the LAPI". We'll only keep adding those tags and then eventually have to remove them all from the test files again, leaving unrelated git changes all over the place. It seems simpler and cleaner to add a `.txt` file where we list running or failing tests.

File: Cython/Utility/TypeConversion.c (hunk 1 above)
Comment (da-woods) [reply Cython/Utility/TypeConversion.c]: OK. I'll drop this part

**Pull request merged**